datafusion/core/src/execution/options.rs

```diff
@@ -377,9 +340,112 @@ impl<'a> NdJsonReadOptions<'a> {
         self.schema = Some(schema);
         self
     }
+}
 
+#[async_trait]
+///
+pub trait ReadOptions<'a> {
     /// Helper to convert these user facing options to `ListingTable` options
-    pub fn to_listing_options(&self, target_partitions: usize) -> ListingOptions {
+    fn to_listing_options(&self, target_partitions: usize) -> ListingOptions;
```

Comment:
```suggestion
    fn to_listing_options(&self, config: &SessionConfig) -> ListingOptions;
```

While I realize that the `target_partitions` is the only thing actually used at the moment, since this is part of the public API I think providing `&SessionConfig` would be more future proof (aka avoid changes to this trait over time)

Reply:
Done!